api/libproject/devstart.rb

```diff
@@ -1589,38 +1589,56 @@ def export_workspace_data(cmd_name, *args)
     :fn => ->(*args) {export_workspace_data("export-workspace-data", *args)}
 })
 
-def generate_impersonated_user_token(cmd_name, *args)
+def generate_impersonated_user_tokens(cmd_name, *args)
   common = Common.new
   ensure_docker cmd_name, args
 
   op = WbOptionsParser.new(cmd_name, args)
   op.add_typed_option(
-      "--output-token-filename [output-token-filename]",
+      "--output-token-filenames [output-token-filename1, ...]",
       String,
-      ->(opts, v) { opts.output_token_filename = v},
-      "Path to an output file for the generated token")
+      ->(opts, v) { opts.output_token_filenames = v},
+      "Comma-separated paths to output file(s) for the generated token(s)")
   op.add_typed_option(
-      "--impersonated-username [impersonated-username]",
+      "--impersonated-usernames [impersonated-username1, ...]",
       String,
-      ->(opts, v) { opts.impersonated_username = v},
-      "AoU researcher email to impersonate, e.g. calbach@fake-research-aou.org")
-  op.add_validator ->(opts) { raise ArgumentError unless (opts.output_token_filename and opts.impersonated_username)}
+      ->(opts, v) { opts.impersonated_usernames = v},
+      "Comma-separated AoU researcher email(s) to impersonate, e.g. calbach@fake-research-aou.org")
+  op.add_validator ->(opts) { raise ArgumentError unless (opts.output_token_filenames and opts.impersonated_usernames)}
   op.parse.validate
 
+  user_email_domain = nil
```

Comment:
there's probably a better to do this with some kind of map and for-all/match-every/whatever